src/main.rs

```diff
@@ -623,8 +623,20 @@ impl Context {
             return Ok(());
         }
 
+        let mut github = if let Some(github) = self.config.github() {
+            github
+        } else {
+            eprintln!("Skipping tagging - GitHub credentials not configured");
+            return Ok(());
+        };
+
         if let Some(repo) = self.config.rustc_tag_repository.clone() {
-            self.tag_repository(signer, &repo, rustc_commit)?;
+            self.tag_repository(signer, &mut github, &repo, rustc_commit)?;
+
+            // Once we've tagged rustc, kick off a thanks workflow run.
+            github
+                .token("rust-lang/thanks")?
```

Comment:
FYI there shouldn't be a need to fetch a new token, as the token is attached to a GitHub App installation, and there can only be one installation of an app for an organization/user. The endpoint that asks for the installation of a repository simply returns the installation of the user/org it's in, and we could reuse the previous token.

Still, it's probably don't worth it to change it in this PR.

Reply:
We could, but we're using this as an opportunity to store the repo name alongside the client. In the future we might also restrict these tokens to just manipulate that repository which gives us some added safety against accidentally pushing to rust-lang/rust master or whatever.